core/whistle_apps-1.0.0/src/whistle_apps_init.erl

```diff
@@ -45,3 +48,57 @@ set_loglevel() ->
     [Error|_] = wh_config:get_atom('log', 'error', ['error']),
     wh_util:change_error_log_level(Error),
     'ok'.
+
+-spec sanity_checks() -> boolean().
+sanity_checks() ->
+    lists:all(fun(F) -> F() end
+              ,[fun does_hostname_resolve_speedily/0
+               ,fun is_system_clock_is_utc/0
+               ]
+             ).
+
+-spec does_hostname_resolve_speedily() -> boolean().
+does_hostname_resolve_speedily() ->
+    InitTime = time_hostname_resolution(),
+    Tests = 20,
+    {Min, Max, Total} =
+        lists:foldl(fun time_hostname_resolution/2
+                   ,{InitTime, InitTime, InitTime}
+                   ,lists:seq(1, Tests)
+                   ),
+    case Max < 5000 of
+        'true' -> 'true';
+        'false' ->
+            lager:warning("hostname results (in us): ~p < ~p < ~p"
+                         ,[Min, (Total div (Tests+1)), Max]
+                         ),
+            lager:critical("hostname resolution is painfully slow!!! This will cause "),
+            'false'
+    end.
+
+-spec time_hostname_resolution(any(), {pos_integer(), pos_integer(), pos_integer()}) ->
```

Comment:
Nah, `wh_now()` is for `erlang:timestamp()`'s return. These are the min, max, and total microseconds it took to resolve the hostname.